sw/device/lib/dif/dif_hmac.c

```diff
@@ -246,11 +246,12 @@ dif_result_t dif_hmac_finish(const dif_hmac_t *hmac,
     return kDifUnavailable;
   }
 
-  // Read the digest.
-  // TODO Static assert register layout.
-  mmio_region_memcpy_from_mmio32(hmac->base_addr, HMAC_DIGEST_0_REG_OFFSET,
-                                 digest->digest,
-                                 HMAC_PARAM_NUM_WORDS * sizeof(uint32_t));
+  // Read the digest in reverse to preserve the numerical value.
+  // The least significant word is at HMAC_DIGEST_7_REG_OFFSET.
+  for (size_t i = 0; i < ARRAYSIZE(digest->digest); ++i) {
+    digest->digest[i] = mmio_region_read32(
+        hmac->base_addr, HMAC_DIGEST_7_REG_OFFSET - i * sizeof(uint32_t));
```

Comment:
Does this change require `kRefExpectedShaDigest` in [hmac_utils](https://github.com/lowRISC/opentitan/blob/fe8a53a299d244d254b4a80eb83d8747d4c046a4/sw/device/lib/testing/hmac_testutils.c#L28-L38) to be updated as well?

Reply:
Thanks for catching this. It looks like we don't run the tests that use `kRefExpectedShaDigest` and `kRefExpectedHmacDigest` in the CI. I'll verify my changes in verilator because the test_rom bitstream seems to be broken because of the `gp` issue.

Reply:
I fixed the key word order as well and verified that `hmac_enc_test` and `hmac_enc_idle_test` pass locally.